sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.students.controller.saveStudentUploadedFiles.e2e-spec.ts

```diff
@@ -0,0 +1,160 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeApplication,
+  saveFakeStudent,
+  saveFakeStudentFileUpload,
+} from "@sims/test-utils";
+import { TestingModule } from "@nestjs/testing";
+import { NotificationActionsService } from "@sims/services";
+import { Raw } from "typeorm";
+import { getDateOnlyFormat } from "@sims/utilities";
+import { FileOriginType, NotificationMessageType } from "@sims/sims-db";
+
+describe("StudentStudentsController(e2e)-saveStudentUploadedFiles", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let appModule: TestingModule;
+  let notificationActionsService: NotificationActionsService;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    notificationActionsService = await module.get(NotificationActionsService);
+    appModule = module;
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("Should save all uploaded files when a proper student file upload request is made.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplication(db.dataSource, { student });
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const studentFile1 = await saveFakeStudentFileUpload(
+      db.dataSource,
+      {
+        student,
+        creator: student.user,
+      },
+      {
+        fileOrigin: FileOriginType.Student,
+        groupName: "application",
+      },
+    );
+    const studentFile2 = await saveFakeStudentFileUpload(
+      db.dataSource,
+      {
+        student,
+        creator: student.user,
+      },
+      {
+        fileOrigin: FileOriginType.Student,
+        groupName: "application",
+      },
+    );
+
+    const payload = {
+      submittedForm: {
+        documentPurpose: "application",
+        applicationNumber: application.applicationNumber,
+      },
+      associatedFiles: [
+        studentFile1.uniqueFileName,
+        studentFile2.uniqueFileName,
+      ],
+    };
+
+    const notificationMessage = await db.notificationMessage.findOne({
+      where: { id: NotificationMessageType.StudentFileUpload },
+    });
+
+    jest
+      .spyOn(
+        notificationActionsService as any,
+        "assertNotificationMessageDetails",
+      )
+      .mockReturnValue({
+        templateId: notificationMessage.templateId,
+        emailContacts: ["test@test.com"],
```

Comment:
I would say that we can save a dummy email for this template at beforeAll instead of mocking.